extensions/openwsman/processors/SourceInitiatedSubscriptionListener.cpp

```diff
@@ -435,7 +435,6 @@ bool SourceInitiatedSubscriptionListener::Handler::handleSubscriptionManager(str
   ws_xml_set_ns(subscription, XML_NS_CUSTOM_SUBSCRIPTION, "m");
 
   // Body/EnumerationResponse/Items/Subscription/Version
-  std::lock_guard<std::mutex> lock(processor_.mutex_);
```

Comment:
This is in an http request handler. Are you certain that civetweb (http server) doesn't use multiple threads that need synchronization on concurrent incoming requests? I don't know it's internals, but I think this might be an issue if it does.

Reply:
It seems to me that the mutex was there because of the member `subscribers_`, and is not related to civetweb. But because I can't tell for sure, and because this processor is unused (AFAIK) I will put it back, it won't do harm. From commit history it seems like nobodoy else touched the processor seriously other than the initial commit from bakaid.

Reply:
My theory is that civetweb calls the handler on a thread that is not in our normal thread pool. This means if the processor is running and an http requests are handled at the same time, or multiple http requests are being handled (on civet threads), then `subscribers_` is accessed concurrently, which is a data race.